src/java/picard/illumina/IlluminaBasecallsToFastq.java

```diff
@@ -454,12 +480,14 @@ public void setInputStream(final InputStream is) {
             reader = new FastqReader(new BufferedReader(new InputStreamReader(is)));
         }
 
+        //TODO: add tests to encode and decode
```

Comment:
I looked at the other SortingCollection<> implementations and found that none of them are tested! Therefore, I'll leave this for a future improvement...